scipy/stats/tests/test_hypotests.py

```diff
@@ -616,6 +619,43 @@ def test_gh_2118(self, x, y, alternative, expected):
                            method="asymptotic")
         assert_allclose(res, expected, rtol=1e-12)
 
+    def teardown(self):
+        _mwu_state._recursive = None
+
+
+class TestMannWhitneyU_iterative(TestMannWhitneyU):
+    def setup(self):
+        _mwu_state._recursive = False
+
+    def teardown(self):
+        _mwu_state._recursive = None
+
+
+@pytest.mark.xslow
+def test_mann_whitney_u_switch():
+    # Check that mannwhiteneyu switches between recursive and iterative
+    # implementations at n = 500
+
+    # ensure that recursion is not enforced
+    _mwu_state._recursive = None
+    _mwu_state._fmnks = -np.ones((1, 1, 1))
```

Comment:
Should I just re-import instead?

Reply:
I think this is cleaner than re-importing.